lib/cachex/services/courier.ex

```diff
@@ -64,7 +69,14 @@ defmodule Cachex.Services.Courier do
   # Due to the nature of the async behaviour, this call will return before
   # the task has been completed, and the :notify callback will receive the
   # results from the task after completion (regardless of outcome).
-  def handle_call({:dispatch, key, task, stack}, caller, {cache, tasks} = state) do
+  def handle_call(
+        {:dispatch, key, task, stack, callers},
+        caller,
+        {cache, tasks} = state
+      ) do
+    put_callers(callers)
+    callers = get_callers()
```

Comment:
I've added the `Courier` process to the cain this way.

The alternative would be to not use the helper and here use: `courier_process = self()`

and then in line 93 do:
```
Process.put(:"$callers", [self() | courier_process | callers])
```

The downside is that `Courier` process itself does not have the callers set, so I decided, it is better to use the helper twice.